src/robot/parsing/lexer/tokenizer.py

```diff
@@ -72,49 +72,52 @@ def _split_from_pipes(self, line):
         yield rest, True
 
     def _cleanup_tokens(self, tokens, data_only):
-        has_data = self._handle_comments(tokens)
-        continues = self._handle_continuation(tokens)
-        self._remove_trailing_empty(tokens)
-        if continues:
-            self._remove_leading_empty(tokens)
-            self._ensure_data_after_continuation(tokens)
-        if data_only:
-            tokens = self._remove_non_data(tokens)
-        return tokens, has_data and not continues
-
-    def _handle_comments(self, tokens):
+        # Handle comments
         has_data = False
-        commented = False
-        for token in tokens:
+
+        iter_in_tokens = iter(tokens)
```

Comment:
2nd improvement was not iterating so many times though the tokens in `_cleanup_tokens`.